modules/core/shared/src/main/scala/coursier/core/Parse.scala

```diff
@@ -21,15 +21,9 @@ object Parse {
         for {
           from <- version(prefix)
           if from.rawItems.nonEmpty
-          last <- Some(from.rawItems.last).collect { case n: Version.Numeric => n }
-          // a bit loose, but should do the job
-          if from.repr.endsWith(last.repr)
-          // appending -a1 to the next version, so has not to include things like
-          // nextVersion-RC1 in the interval - nothing like nextVersion* should be included
-          to <- version(from.repr.stripSuffix(last.repr) + last.next.repr + "-a1")
+          to <- version(prefix + "-max")
           // the contrary would mean something went wrong in the loose substitution above
-          if from.rawItems.init == to.rawItems.dropRight(2).init
-          if to.rawItems.takeRight(2) == Seq(Version.Literal("a"), Version.Number(1))
+          if from.rawItems.init == to.rawItems.dropRight(1).init
         } yield VersionInterval(Some(from), Some(to), fromIncluded = true, toIncluded = false)
```

Comment:
```suggestion
         } yield VersionInterval(Some(from), Some(to), fromIncluded = true, toIncluded = true)
```
Change `toIncluded` to `true`? So that the version ending with `-max` falls into the interval too (`1.2-max` should match `1.2+` I guess).

Reply:
I thought `max` is not a legit component of a version, rather it is a placeholder for a "maximal number" in a version range / spec? So you should never actually see a version `1.2-max`...

Reply:
Still, with the former behavior, these `-max` versions were in the interval. Let's keep them there just in case for now.

Reply:
Fine, I'll change that back.